### PR TITLE
Sub-issue (#2287): APU CH1 sweep — Phases 4-6 deferred recalc machinery (Phase 7 deferred)

### PR DESCRIPTION
**Pull request by rmstdope**
## Summary

Implements the SameBoy-derived deferred sweep recalculation machinery for CGB-E (Phases 4-6 of #2287), gated behind an internal disabled flag while SameSuite ROM parity (Phase 7) is pending follow-up.

Refs #2287 #2277.

### Phase 4 — deferred recalc
- Split `clock_sweep` into arm + drain: `arm_sweep_calculation`, `sweep_calculation_done`, `sweep_tick` (1 MHz).
- New fields: `sweep_calc_countdown`, `sweep_calc_reload_timer`, `sweep_length_addend`, `completed_addend`, `unshifted_sweep`, `instant_calc_done`, `sweep_tick_phase`.

### Phase 5 — dual overflow
- Trigger keeps the synchronous overflow check (Blargg-aligned).
- `sweep_calculation_done` performs the second (mid-sweep) overflow check via `shadow + sweep_length_addend`, with `restart_hold` gating the shadow refresh.

### Phase 6 — NR10 SameBoy semantics
- `write_nr10` splits into `write_nr10_synchronous` (DMG / pre-CGB-E) and `write_nr10_deferred` (CGB-E).
- Deferred path uses SameBoy's disable formula: `shadow + completed_addend + old_negate > 0x7FF` AND new value clears negate bit.
- `nr10_write_glitch` corrupts in-flight sub-cycle countdowns when NR10 is rewritten mid-flight.
- Re-arm path when `sweep_countdown == 7`.

## Routing decision

`uses_deferred_sweep()` currently returns `false` unconditionally. The deferred path is fully implemented but disabled in production because neither Blargg sweep ROMs nor SameSuite sweep ROMs pass with it active — additional CGB-E quirks are missing. A test-only override (`force_deferred_sweep_for_tests`, behind `#[cfg(test)]`) lets unit tests still exercise the deferred path.

This shipped state therefore preserves Phase 0-3 behaviour for end users (DMG + CGB Blargg ROMs continue to pass) while landing the architectural scaffolding needed for Phase 7.

## Phase 7 — deferred to follow-up sub-issue

The 3 SameSuite sweep ROMs (`channel_1_sweep`, `channel_1_sweep_restart`, `channel_1_sweep_restart_2`) all return code `B` (0x42) with the deferred path active. Their `#[ignore]` annotations are updated to reflect the new status. A follow-up sub-issue should:
1. Flip `uses_deferred_sweep()` to `is_cgb && cgb_model == CgbE`.
2. Identify and implement the missing CGB-E sweep quirks (likely candidates: `during_div_write` interaction, pre-CGB-D `old_negate` force, `restart_hold` edge cases).
3. Verify Blargg CGB sweep ROMs still pass.

## Tests

- 11 new `channel1` unit tests covering Phases 4-6 deferred semantics (gated to deferred path via test-only override).
- All 9343 `cargo test --no-default-features --lib` tests pass.
- `cargo clippy --all-targets --all-features -- -D warnings` clean.
- `cargo fmt` clean.
- `wasm-pack test --headless --chrome --no-default-features --features wasm` — 54 passed.
- Python `unittest` — 223 passed.
- `npm test` — 298 passed in 37 files.

## Definition of Done checklist

- [x] Architectural Phase 4-6 work implemented.
- [x] Unit tests cover new deferred semantics.
- [x] No regressions vs origin/main (DMG/CGB Blargg sweep all green).
- [ ] Phase 7 ROM parity — deferred to follow-up sub-issue.
- [x] Pre-merge checks all pass.

### CODE REVIEW

File: src/gb/apu/channel1.rs [hunk 1]
```diff
@@ -269,9 +347,95 @@ impl Channel1 {
...
+        // SameBoy: sample_length = sweep_length_addend + shadow + (NR10 & 8 ? 1 : 0); &= 0x7FF.
+        if self.sweep_shift > 0 {
+            let neg_bit: u16 = if self.sweep_negate { 1 } else { 0 };
+            self.freq = self
+                .sweep_length_addend
+                .wrapping_add(self.sweep_shadow)
+                .wrapping_add(neg_bit)
+                & 0x7FF;
+            trace_apu!(3; "GB APU CH1 sweep writeback shadow=0x{:03X} addend=0x{:03X} freq=0x{:03X}",
+                self.sweep_shadow, self.sweep_length_addend, self.freq);
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/gb/apu/channel1.rs]: Addressed in 6a249cff: writeback now uses completed_addend (set at the end of the previous sweep_calculation_done) so a re-arm mid-flight cannot apply a stale recomputed addend. Updated the trace to log completed_addend and revised the surrounding comment.

File: src/gb/apu/apu.rs [hunk 2]
```diff
@@ -288,6 +288,12 @@ impl Apu {
         self.ch3.tick();
         self.ch4.tick();
 
+        // ── CH1 sweep machinery (1 MHz cadence; gated internally) ─────────
+        // SameBoy `Core/apu.c` `GB_apu_run` advances sweep state at half the
+        // M-cycle rate; `Channel1::sweep_tick` toggles its own phase flag and
+        // performs work every other call.
+        self.ch1.sweep_tick();
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/gb/apu/apu.rs:295]: Addressed in 6a249cff: sweep_tick now early-returns when sweep_calc_reload_timer == 0 and sweep_calc_countdown == 0 and instant_calc_done is false, before toggling sweep_tick_phase. The hot path is effectively free in production where the deferred machinery is never armed, and serialized state stops being mutated every M-cycle.
